frontend/src/main/components/Users/UsersTable.js

```diff
@@ -29,11 +31,75 @@ const columns = [
         id: 'admin',
         accessor: (row, _rowIndex) => String(row.admin) // hack needed for boolean values to show up
     },
+    {
+        Header: 'Instructor',
+        id: 'instructor',
+        accessor: (row, _rowIndex) => String(row.instructor) // hack needed for boolean values to show up
+    }
 ];
 
-export default function UsersTable({ users }) {
+// function cellToAxiosParamsToggleAdmin(cell){
+//     return{
+//         url: "/api/admin/users/toggleAdmin",
+//         method: "POST",
+//         params:{
+//             id: cell.row.values.id
+//         }
+//     }
+// }
+
+// function cellToAxiosParamsToggleInstructor(cell){
+//     return{
+//         url: "/api/admin/users/toggleInstructor",
+//         method: "POST",
+//         params: {
+//             id: cell.row.values.id
+//         }
+//     }
+// }
+
+export default function UsersTable({ users, showToggleButtons }) {
+    // Stryker disable all : hard to test for query caching
+    const toggleAdminMutation = useBackendMutation(
+        cellToAxiosParamsToggleAdmin,
+        {onSuccess: toggleAdminSuccess},
+        ["/api/admin/users"]
+    );
+    // Stryker restore all 
+
+    // Stryker disable next-line all : TODO try to make a good test for this
+    const toggleAdminCallback = async (cell) => {toggleAdminMutation.mutate(cell);}
+
+    // toggle instructor
+    // Stryker disable all : hard to test for query caching
+    const toggleInstructorMutation = useBackendMutation(
+        cellToAxiosParamsToggleInstructor,
+        {onSuccess: toggleInstructorSuccess},
+        ["/api/admin/users"]
+    );
+    // Stryker restore all 
+
+    // Stryker disable next-line all : TODO try to make a good test for this
+    const toggleInstructorCallback = async (cell) => {toggleInstructorMutation.mutate(cell);}
+
+    const buttonColumns = [
+        ...columns,
+        ButtonColumn("toggle-admin", "primary", toggleAdminCallback, "UsersTable"),
+        ButtonColumn("toggle-instructor", "primary", toggleInstructorCallback, "UsersTable")
+    ]
+
+    var retColumns;
+    if (showToggleButtons){
+        retColumns = buttonColumns;
+    }else{
+        retColumns = columns;
+    }
+
     return <OurTable
         data={users}
-        columns={columns}
+        showToggleButtons={showToggleButtons}
```

Comment:
Why is this being passed in?
```suggestion
```

Reply:
I agree with @chriscerie ... there doesn't appear to be any usage of `showToggleButtons` inside the `OurTable` component (see: https://github.com/ucsb-cs156-f23/proj-organic-f23-6pm-3/blob/main/frontend/src/main/components/OurTable.js )

So I think you may want to accept Chris' suggestion here as well.